source/common/grpc/async_client_impl.h

```diff
@@ -257,17 +259,47 @@ class AsyncStreamImpl : public AsyncStream<RequestType>,
   friend class AsyncClientImpl<RequestType, ResponseType>;
 };
 
+class AsyncClientTracingConfig : public Tracing::Config {
+public:
+  // Tracing::Config
+  Tracing::OperationName operationName() const override { return Tracing::OperationName::Egress; }
+  const std::vector<Http::LowerCaseString>& requestHeadersForTags() const override {
+    return request_headers_for_tags_;
+  }
+
+  struct {
+    const std::string STATUS = "status";
+    const std::string GRPC_STATUS = "grpc.status_code";
+    const std::string CANCELED = "canceled";
+    const std::string ERROR = "error";
+    const std::string TRUE = "true";
+    const std::string UPSTREAM_CLUSTER_NAME = "upstream_cluster_name";
+  } TagStrings;
+
+private:
+  const std::vector<Http::LowerCaseString> request_headers_for_tags_;
+};
+
+typedef ConstSingleton<AsyncClientTracingConfig> TracingConfig;
+
 template <class RequestType, class ResponseType>
 class AsyncRequestImpl : public AsyncRequest,
                          public AsyncStreamImpl<RequestType, ResponseType>,
                          AsyncStreamCallbacks<ResponseType> {
 public:
   AsyncRequestImpl(AsyncClientImpl<RequestType, ResponseType>& parent,
                    const Protobuf::MethodDescriptor& service_method, const RequestType& request,
-                   AsyncRequestCallbacks<ResponseType>& callbacks,
+                   AsyncRequestCallbacks<ResponseType>& callbacks, Tracing::Span& parent_span,
                    const Optional<std::chrono::milliseconds>& timeout)
       : AsyncStreamImpl<RequestType, ResponseType>(parent, service_method, *this, timeout),
-        request_(request), callbacks_(callbacks) {}
+        request_(request), callbacks_(callbacks) {
+
+    current_span_ = parent_span.spawnChild(TracingConfig::get(),
+                                           "async " + parent.remote_cluster_name_ + " egress",
+                                           ProdSystemTimeSource::instance_.currentTime());
```

Comment:
reminds me that we need to upgrade lightstep to add monotonic time to tracing

Reply:
Yeah we do need to do this. I think that entire driving is being worked on so hopefully we can get it swapped out with the OT driver update.